### PR TITLE
[C#] Add buildTransitive directory to NuGet package

### LINKED ISSUE #25285: [C#] grpc_csharp_ext.x86.dll and grpc_csharp_ext.x64.dll not copied to output in netstandard2.0
### Is your feature request related to a problem? Please describe.

This request is specifically for the Grpc.Core package on nuget: https://www.nuget.org/packages/Grpc.Core/2.35.0

We recently transitioned to netstandard2.0 and encountered errors because grpc_csharp_ext.x86.dll and grpc_csharp_ext.x64.dll were not found in output directory. After a bit of investigating, we found that the build step to copy these files only exists in net45 and not netstandard2.0 (Or any other versions). We are currently doing it manually to ensure our projects actually work, but this feels like a bug as in the net45 targets file it explicitly states that users should only do it manually if they really know what they are doing and in netstandard2.0 they have to do it themselves. It is also undesirable as it could cause further issues if the file structure changes in a future update, etc.

Are there any plans to update the build steps for the nuget packages?

### Describe the solution you'd like

For Grpc.Core to add the same copy to output directory build step for netstandard2.0 as well

### Describe alternatives you've considered

We are currently doing the output manually but this is undesirable for many reasons. Including future directory tree modification could cause the files to not be found.

### Additional context
Add any other context about the feature request here.

image

### PR DESCRIPTION
**Pull request by dfederm**
This change adds the buildTransitive directory to the NuGet package. This allows projects which depend on projects which depend on Grpc.Core, but don't build on Grpc.Core directly, to receive the build logic from Grpc.Core, which is required to copy native runtime dependencies for .NET Framework projects.

Fixes #25285

@jtattermusch

### CODE REVIEW

**Comment by jtattermusch** [comment]
Adhoc distribtests run https://fusion2.corp.google.com/invocations/1e6d8e9c-5778-4a58-9ed1-a636852d965e (will post back results later).

**Comment by jtattermusch** [comment]
@apolcyn  please another googler's LGTM.

**Review by jtattermusch — APPROVED** [review]
LGTM after the tests pass. Can you confirm that if .targets from both build and buildTransitive get included, this wouldn't result in any warnings / issues with the build? (see comment)

File: src/csharp/Grpc.Core/buildTransitive/net45/Grpc.Core.targets
Comment (dfederm) [reply src/csharp/Grpc.Core/buildTransitive/net45/Grpc.Core.targets:5]: It doesn't look like it is included twice. I just tested by putting an explicit reference to Grpc.Core in the console project and I don't see a double-import.

Furthermore, upon inspecting the console app's project.assets.json, I see only buildTransitive is included:

```json
    ".NETFramework,Version=v4.7.2/win7-x86": {
      "Grpc.Core/2.35.0": {
        "type": "package",
        "dependencies": {
          "Grpc.Core.Api": "2.35.0",
          "System.Memory": "4.5.3"
        },
        "frameworkAssemblies": [
          "Microsoft.CSharp",
          "System"
        ],
        "compile": {
          "lib/net45/Grpc.Core.dll": {}
        },
        "runtime": {
          "lib/net45/Grpc.Core.dll": {}
        },
        "native": {
          "runtimes/win-x86/native/grpc_csharp_ext.x86.dll": {}
        },
        "build": {
          "buildTransitive/net45/Grpc.Core.targets": {}
        }
      },
```

So it seems that buildTransitive takes precedence.

**Comment by jtattermusch** [comment]
The distribtests are green (except one test that was known to be broken for an unrelated reason) 
https://fusion2.corp.google.com/invocations/7d96f585-e0bf-4d7d-9e9f-4a154df9195a

**Comment by jtattermusch** [comment]
This can be merged once @apolcyn  approves.

**Review by apolcyn — APPROVED** [review]

**Pull request merged**
